### PR TITLE
Migration to Vercel

### PR DESCRIPTION
**Pull request by natapg**
Repo migration

- [ ]  Warn the Marketing team not to use the CMS until the process is finished
- [ ]  Freeze on Netlify the latest production version
- [ ]  Change the Vercel project repository to use the original repository instead of the fork (deploy will fail) (only Vercel admin can do this)
- [ ]  Remove from the status checks the Netlify deployment from the original repository (only admin can do it). This is necessary to be able to merge the PR to be created
- [x]  Create a PR from the fork to the main repo → Vercel Preview should build without issues / Netlify Deploy Preview will probably fail
- [ ]  Give the GitHub app (created by the organization admin) permissions to the repository
- [ ]  Add people who use the CMS as repository collaborators
- [ ]  Merge fork
- [ ]  Test CMS, should create a commit on master and build on Vercel (Netlify should be blocked and should not change the production version)
- [ ]  Test CI (lint, prettier, test, lighthouse, bundlewatch, storybook)
- [ ]  If everything went well, change the CloudFront source to the Vercel domain instead of Netlify
- [ ]  If everything went wrong, revert the last commit and continue using Netlify until the problem is solved
- [ ]  Add Vercel deploy to required statuses
- [ ]  Remove Netlify projects ([fingeprintjs.com](http://fingeprintjs.com/) and [fingerprint.com](http://fingerprint.com/))

### CODE REVIEW

File: static/admin/config.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 backend:
-  name: git-gateway
+  name: github
+  repo: fingerprintjs/fingerprint.com-vercel
```
Comment (natapg) [inline static/admin/config.yml:3]: reminder to change it to fingerprintjs/fingerprint.com

File: src/constants/env.ts [hunk 2]
```diff
@@ -1,117 +1,38 @@
-type EnvironmentVariable = string | number | boolean | undefined
-
-enum NetlifyContext {
```
Comment (natapg) [inline src/constants/env.ts:3]: Vercel allows as to assign different values to the same variable depending on the environment

File: src/node/social-card/generate-image.js [hunk 3]
```diff
@@ -1,11 +1,7 @@
-const { registerFont, createCanvas, loadImage } = require('canvas')
+const { createCanvas, loadImage } = require('canvas')
 const { join } = require('path')
 
 exports.generateImage = async (title) => {
-  registerFont(join('src', 'node', 'social-card', 'assets', 'Inter-SemiBold.ttf'), {
```
Comment (natapg) [inline src/node/social-card/generate-image.js:5]: This command doesn't work during build in Vercel, so I add the font directly to the system

File: gatsby-node.js [hunk 4]
```diff
@@ -369,10 +380,7 @@ exports.sourceNodes = async ({ actions, getNodes, createContentDigest }) => {
     prepaidQuantity: Number(defaultPrepaidQuantity),
   }
 
-  const apiUrl =
-    process.env.CONTEXT === 'production'
-      ? process.env.GATSBY_FPJS_MGMT_API_HOST
-      : process.env.GATSBY_PREVIEW_FPJS_MGMT_API_HOST
+  const apiUrl = process.env.GATSBY_FPJS_MGMT_API_HOST
```
Comment (tan-90) [inline gatsby-node.js:383]: I assume vercel is smart enough to change the variable based on whether or not it's a preview?

File: src/constants/env.ts (hunk 2 above)
Comment (tan-90) [reply src/constants/env.ts:3]: Answers my  question from 2 seconds ago haha

File: static/admin/config.yml (hunk 1 above)
Comment (tan-90) [reply static/admin/config.yml:3]: :feels-strong:

File: setup.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+make
+cp libz.so.1.2.9 ../node_modules/canvas/build/Release/libz.so.X
+cd ..
+
+patchelf --replace-needed /lib64/libz.so.1 libz.so.X ./node_modules/canvas/build/Release/libpng16.so.16
+patchelf --replace-needed libz.so.1 libz.so.X ./node_modules/canvas/build/Release/libpng16.so.16
+
+# there is a problem with the registerFonts command, so we install the font on the system
+mkdir ~/.local/share/fonts
+mv Inter-SemiBold.ttf ~/.local/share/fonts/Inter-SemiBold.ttf
```
Comment (tan-90) [inline setup.sh:26]: Nit: missing blank line

File: setup.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+cd zlib-1.2.9
+sh configure
+make
+cp libz.so.1.2.9 ../node_modules/canvas/build/Release/libz.so.X
+cd ..
+
+patchelf --replace-needed /lib64/libz.so.1 libz.so.X ./node_modules/canvas/build/Release/libpng16.so.16
+patchelf --replace-needed libz.so.1 libz.so.X ./node_modules/canvas/build/Release/libpng16.so.16
+
+# there is a problem with the registerFonts command, so we install the font on the system
```
Comment (tan-90) [inline setup.sh:24]: :gigachad:

**Review by tan-90 — APPROVED** [review]

**Pull request closed without merging**